File: 프로그래머스 Lv2/k진수에서 소수 개수 구하기.swift
import Foundation
func converseNumber(_ n:Int, _ k: Int) -> String {
    var reverseBase: String = ""
    var quotient: Int = n
    var remainder: Int

    while quotient > 0 {
        (quotient, remainder) = quotient.quotientAndRemainder(dividingBy: k)
        reverseBase += String(remainder)
    }

    return String(reverseBase.reversed())
}
func isPrimeNumber(_ n:Int) -> Bool {
    if n == 1 { return false }
    for i in 2..<Int(sqrt(Double(n))) + 1 {
        if n % i == 0 { return false }
    }
    return true
}

func solution(_ n:Int, _ k:Int) -> Int {
    
    var answer = 0
    let strNumber = converseNumber(n, k)
    let candidates = strNumber.split(separator: "0").map { Int($0)! }
    
    for c in candidates {
        if isPrimeNumber(c) { answer += 1 }
    }
    
    return answer
}

print(solution(437674, 3))



import Foundation

func isPrimeNumber(_ n:Int) -> Bool {
    if n == 1 { return false }
    for i in 2..<Int(sqrt(Double(n))) + 1 {
        if n % i == 0 { return false }
    }
    return true
}

func solution(_ n:Int, _ k:Int) -> Int {
    
    var answer = 0
    let strNumber = String(n, radix: k)
    let candidates = strNumber.split(separator: "0").map { Int($0)! }
    
    for c in candidates {
        if isPrimeNumber(c) { answer += 1 }
    }
    
    return answer
}

print(solution(437674, 3))

File: 프로그래머스 Lv2/k진수에서 소수 개수 구하기.py
import math

def is_prime_number(x):
    if x == 1:
        return False
    for i in range(2, int(math.sqrt(x)) + 1):
        if x % i == 0:
            return False
    return True

def converse_number(n, q):
    rev_base = ''

    while n > 0:
        n, mod = divmod(n, q)
        rev_base += str(mod)

    return rev_base[::-1]

def solution(n, k):
    answer = 0

    number = converse_number(n, k)
    candidates = number.split("0")

    for c in candidates:
        try:
            if c == "0":
                continue
            else:
                if is_prime_number(int(c)):
                    answer += 1
        except:
            continue

    return answer


print(solution(110011,	10))
